### PR TITLE
LPD-29137 - Format workspaces and playwright with node-scripts

### PR DESCRIPTION
**Pull request by bryceosterhaus**

### CODE REVIEW

File: modules/_node-scripts/bin.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ const COMMANDS = {
 	'format': {
 		description:
 			'formats source files or optionally only checks with "--check" flag.',
-		parameters: '[--check]',
+		parameters: '[--all, --check]',
```
Comment (izaera) [inline modules/_node-scripts/bin.js:35]: This should be something like `[--all] [--check]` if both are optional and both can be given at the same time.

`[{--all|--check}]` if they are optional but exclusive, etc

Based on the informal unix/DOS command line syntax documentation

**Comment by bryceosterhaus** [comment]
ci:test:sf

File: modules/_node-scripts/format/format.mjs [hunk 2]
```diff
@@ -29,17 +29,19 @@ async function getFilesToCheck(rootDir) {
 	const prettierIgnoreFilePath = path.join(rootDir, PRETTIER_IGNORE_FILE);
 	const gitIgnoreFilePath = path.join(rootDir, GIT_IGNORE_FILE);
 
-	const eslintIgnores = readIgnoreFile(eslintIgnoreFilePath);
-	const prettierIgnores = readIgnoreFile(prettierIgnoreFilePath);
-	const gitIgnores = readIgnoreFile(gitIgnoreFilePath);
+	const eslintIgnores = await readIgnoreFile(eslintIgnoreFilePath);
```
Comment (izaera) [inline modules/_node-scripts/format/format.mjs:32]: You should read these three in parallel with Promise.all

**Comment by bryceosterhaus** [comment]
ci:test:relevant

File: modules/_node-scripts/util/filterChangedFiles.mjs [hunk 3]
```diff
@@ -23,15 +23,28 @@ import path from 'path';
...
+			);
+		}
 
-	if (upstream === undefined) {
 		return files;
 	}
 
-	const {stdout: topLevel} = await $`git rev-parse --show-toplevel`;
+	console.log(
+		`Only running against files changed between '${upstream}' and '${currentBranch}' branches. Use flag '--all' to run against all files.`
```
Comment (izaera) [inline modules/_node-scripts/util/filterChangedFiles.mjs:44]: Missing an ℹ️ or ⚠️ emoji here :joy:

File: modules/_node-scripts/util/getNamedArguments.mjs [hunk 4]
```diff
@@ -10,8 +10,19 @@ export default function getNamedArguments(argumentNames) {
 
 	while (argv.length) {
 		for (const [argName, argSwitch] of Object.entries(argumentNames)) {
-			if (argv[0].startsWith(`${argSwitch}=`)) {
-				namedArgs[argName] = argv[0].substring(argSwitch.length + 1);
+			if (argv[0].startsWith(argSwitch)) {
```
Comment (izaera) [inline modules/_node-scripts/util/getNamedArguments.mjs:13]: :ok_hand:

**Review by izaera — APPROVED** [review]
I left some comments, but LGTM

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:report:22172300

**Comment by bryceosterhaus** [comment]
ci:forward:force

**Comment by bryceosterhaus** [comment]
ci:forward:force

**Pull request closed without merging**